unity-renderer/Assets/DCLPlugins/ECS7/Systems/CameraSystem/ECSCameraEntitySystem.cs

```diff
@@ -9,81 +7,92 @@
 using UnityEngine;
 
 namespace ECSSystems.CameraSystem
 {
-    public static class ECSCameraEntitySystem
+    public class ECSCameraEntitySystem : IDisposable
     {
         private static readonly PBCameraMode reusableCameraMode = new PBCameraMode();
         private static readonly PBPointerLock reusablePointerLock = new PBPointerLock();
 
-        private class State
+        private readonly BaseVariable<Transform> cameraTransform;
+        private readonly RendererState rendererState;
+        private readonly Vector3Variable worldOffset;
+        private readonly BaseList<IParcelScene> loadedScenes;
+        private readonly IECSComponentWriter componentsWriter;
+        private readonly BaseVariableAsset<CameraMode.ModeId> cameraMode;
+
+        private Vector3 lastCameraPosition = Vector3.zero;
+        private Quaternion lastCameraRotation = Quaternion.identity;
+        private bool newSceneAdded = false;
+
+        public ECSCameraEntitySystem(IECSComponentWriter componentsWriter)
         {
-            public BaseVariable<Transform> cameraTransform;
-            public RendererState rendererState;
-            public Vector3Variable worldOffset;
-            public IReadOnlyList<IParcelScene> loadedScenes;
-            public IECSComponentWriter componentsWriter;
-            public BaseVariableAsset<CameraMode.ModeId> cameraMode;
-            public UnityEngine.Vector3 lastCameraPosition = UnityEngine.Vector3.zero;
-            public Quaternion lastCameraRotation = Quaternion.identity;
+            cameraTransform = DataStore.i.camera.transform;
+            rendererState = CommonScriptableObjects.rendererState;
+            worldOffset = CommonScriptableObjects.worldOffset;
+            loadedScenes = DataStore.i.ecs7.scenes;
+            this.componentsWriter = componentsWriter;
+            cameraMode = CommonScriptableObjects.cameraMode;
+
+            loadedScenes.OnAdded += LoadedScenesOnOnAdded;
         }
 
-        public static Action CreateSystem(IECSComponentWriter componentsWriter)
+        public void Dispose()
         {
-            var state = new State()
-            {
-                cameraTransform = DataStore.i.camera.transform,
-                rendererState = CommonScriptableObjects.rendererState,
-                worldOffset = CommonScriptableObjects.worldOffset,
-                loadedScenes = DataStore.i.ecs7.scenes,
-                componentsWriter = componentsWriter,
-                cameraMode = CommonScriptableObjects.cameraMode
-            };
-            return () => Update(state);
+            loadedScenes.OnAdded -= LoadedScenesOnOnAdded;
         }
 
-        private static void Update(State state)
+        public void Update()
         {
-            if (!state.rendererState.Get())
+            if (!rendererState.Get())
                 return;
 
-            Transform cameraT = state.cameraTransform.Get();
+            Transform cameraT = cameraTransform.Get();
```

Comment:
I believe the proposal contradicts a bit with the ECS principle of systems and a data driven approach in general. And it just returns a value, so a minimum overhead, is not it?

Reply:
Isnt it still data driven? you are listening to change in the data in the end.

Reply:
I think that could be accomplished but it would need a refactor of the whole `ECSCameraEntitySystem` to stop using `Update()` and instead respond based on the Camera Transform `OnChange()` event, and that probably conflicts with the goal of [these "LateUpdate systems"](https://github.com/decentraland/unity-renderer/blob/ed55888535e4479c767fc967a11761e81ea1f21b/unity-renderer/Assets/DCLPlugins/ECS7/Systems/ECSSystemsController.cs#L79~L83) (the same would apply to the `ECSPlayerTransformSystem` and its usage of `avatarTransform.Get()`).

I will take note on this and discuss it with @pbosio and the SDK team after he returns from PTO next week, to see if it makes sense to implement that refactor (in another issue+PR), meanwhile if we can avoid holding the PR due to this it would be great.

Reply:
No refactor for that  nitpicking comment hahaha, it is fine as it is.